Test/HCThread_Internal.c

```diff
@@ -0,0 +1,31 @@
+//
+//  HCThread_Internal.c
```

Comment:
_Internal.c? We don't ever need an internal implementation file, do we? This should just be in HCThread.c

Reply:
I am testing implementation details and internal aspects is why I ended up doing it this way. I can move it if you want.